### PR TITLE
Improve top level crate documentation

### PR DESCRIPTION
**Pull request by n1ghtmare**
Added and rephrased some documentation.

### CODE REVIEW

File: async-nats/src/lib.rs [hunk 1]
```diff
@@ -11,41 +11,50 @@
...
 //! use bytes::Bytes;
 //! use futures::StreamExt;
 //!
 //! #[tokio::main]
 //! async fn main() -> Result<(), async_nats::Error> {
+//!     // Connect to the NATS server
 //!     let client = async_nats::connect("demo.nats.io").await?;
+//!
+//!     // Subscribe to the "messages" subject and limit to 10 messages
 //!     let mut subscriber = client.subscribe("messages".into()).await?.take(10);
```
Comment (Jarema) [inline async-nats/src/lib.rs:50]: Remove the `take` - it's more confusing than helping.

**Review by Jarema — CHANGES_REQUESTED** [review]
one small change request.

File: async-nats/src/lib.rs (hunk 1 above)
Comment (n1ghtmare) [reply async-nats/src/lib.rs]: Agreed! Also, done!

**Review by caspervonb — CHANGES_REQUESTED** [review]

File: async-nats/src/lib.rs [hunk 2]
```diff
@@ -11,41 +11,50 @@
...
-//! system for cloud native applications, `IoT` messaging, and microservices
+//! ```bash
+//! git clone https://github.com/nats-io/nats.rs
+//! ````
+//! NATS.io is a simple, secure, and high-performance open-source messaging
+//! system designed for cloud-native applications, IoT messaging, and microservices
 //! architectures.
 //!
-//! For sync API refer <https://crates.io/crates/nats>
+//! For the synchronous API, you can refer to:
```
Comment (Jarema) [inline async-nats/src/lib.rs:25]: Mention here that sync api is deprecated (it is not yet, but will be very soon, and no one looking at async repo should have ideas to go there.

File: async-nats/src/lib.rs [hunk 3]
```diff
@@ -11,41 +11,50 @@
...
-//! [https://nats.io/]: https://nats.io/
+//! For more information on NATS.io visit: <https://nats.io>
 //!
 //! ## Examples
 //!
-//! Below you can find some basic examples how to use this library.
+//! Below, you can find some basic examples on how to use this library.
 //!
-//! For details, refer docs for specific method/struct.
+//! For more details, please refer to the documentation for the specific methods and structures.
```
Comment (Jarema) [inline async-nats/src/lib.rs:34]:
```suggestion
//! For more details, please refer to the specific methods and structures documentation.
```

**Review by Jarema — CHANGES_REQUESTED** [review]
Some nits.

**Review by Jarema — CHANGES_REQUESTED** [review]
It does not pass the spelling test.

File: async-nats/src/lib.rs [hunk 4]
```diff
@@ -11,41 +11,50 @@
...
+//! Below, you can find some basic examples on how to use this library.
 //!
-//! For details, refer docs for specific method/struct.
+//! For more details, please refer to the specific methods and structures documentation.
 //!
 //! ### Complete example
 //!
+//! Connect to the NATS server, publish messages and subscribe to receive messages.
+//!
 //! ```
```
Comment (Jarema) [inline async-nats/src/lib.rs:40]:
```suggestion
//! ```no_run
```

**Review by Jarema — CHANGES_REQUESTED** [review]
doc test deadlocks.

**Review by Jarema — APPROVED** [review]
LGTM

**Review by caspervonb — APPROVED** [review]

**Pull request merged**